File: pytools/examples/golang/src/client/proto/msg_role_base.go
package proto

import (
	"packet"
)

type MsgRoleBase struct {
	uid                      uint32
	uname                    string
}

func (this *MsgRoleBase) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint32(this.uid)
	pack.WriteString(this.uname)

	return pack.ReadBytes()
}

func MsgRoleBaseDecode(pack *packet.Packet) *MsgRoleBase {
	msgRoleBase := &MsgRoleBase{}

	msgRoleBase.uid = pack.ReadUint32()
	msgRoleBase.uname = pack.ReadString()
	return msgRoleBase
}

func (this *MsgRoleBase) SetUid(uid uint32) {
	this.uid = uid
}

func (this *MsgRoleBase) GetUid() uint32 {
	return this.uid
}

func (this *MsgRoleBase) SetUname(uname string) {
	this.uname = uname
}

func (this *MsgRoleBase) GetUname() string {
	return this.uname
}

File: pytools/data/data.code.cpp.client/MsgRoleBase.cpp
#ifndef _MSG_ROLE_BASE_
#define _MSG_ROLE_BASE_

#include <list>

#include <pb_type.hpp>
#include <Packet.hpp>

#include <Msg.hpp>



class MsgRoleBase
{
private:
	U32 m_uid;
	String m_uname;


public:
	BYTE* Encode(U32* len)
	{
		Packet packet;
		packet.WriteU32(m_uid);
		packet.WriteString(m_uname);
		return packet.ReadBytes(len);
	}

	MsgRoleBase()
	{

	}

	MsgRoleBase(Packet* packet)
	{
		m_uid = packet->ReadU32();
		m_uname = packet->ReadString();
	}

	BYTE* GetBytes(U32* len)
	{
		return Encode(len);
	}


	void SetUid(U32 uid)
	{
		m_uid = uid;
	}
	U32 GetUid()
	{
		return m_uid;
	}

	void SetUname(String uname)
	{
		m_uname = uname;
	}
	String GetUname()
	{
		return m_uname;
	}

};

#endif
